data-prepper-plugins/cloudwatch-logs/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/cloudwatch_logs/CouldWatchLogsIT.java

```diff
@@ -0,0 +1,276 @@
+                    assertThat(events.size(), equalTo(2));
+                    for (int i = 0; i < events.size(); i++) {
+                        String message = events.get(i).message();
+                        Map<String, Object> event = objectMapper.readValue(message, Map.class);
+                        assertThat(event.get("name"), equalTo("Person"+i));
+                        assertThat(event.get("age"), equalTo(Integer.toString(i)));
+                    }
+                });
+
+    }
+
+    @Test
+    void TestSinkOperationWithMaxRequestSize() throws Exception {
+        long startTime = Instant.now().toEpochMilli();
+        when(thresholdConfig.getBatchSize()).thenReturn(20);
+        when(thresholdConfig.getMaxRequestSizeBytes()).thenReturn(108L);
+        
+        sink = createObjectUnderTest();
+        Collection<Record<Event>> records = getRecordList(2);
+        sink.doOutput(records);
+        await().atMost(Duration.ofSeconds(30))
+                .untilAsserted(() -> {
+                    long endTime = Instant.now().toEpochMilli();
+                    GetLogEventsRequest getRequest = GetLogEventsRequest
+                                       .builder()
+                                       .logGroupName(logGroupName)
+                                       .logStreamName(logStreamName)
+                                       .startTime(startTime)
+                                       .endTime(endTime)
+                                       .build();
+                    GetLogEventsResponse response = cloudWatchLogsClient.getLogEvents(getRequest);
+                    List<OutputLogEvent> events = response.events();
+                    assertThat(events.size(), equalTo(2));
+                    for (int i = 0; i < events.size(); i++) {
+                        String message = events.get(i).message();
+                        Map<String, Object> event = objectMapper.readValue(message, Map.class);
+                        assertThat(event.get("name"), equalTo("Person"+i));
+                        assertThat(event.get("age"), equalTo(Integer.toString(i)));
+                    }
+                });
+
+    }
+
+    private Collection<Record<Event>> getRecordList(int numberOfRecords) {
+        final Collection<Record<Event>> recordList = new ArrayList<>();
+        List<HashMap> records = generateRecords(numberOfRecords);
+        for (int i = 0; i < numberOfRecords; i++) {
+            final Event event = JacksonLog.builder().withData(records.get(i)).build();
+            recordList.add(new Record<>(event));
+        }
+        return recordList;
+    }
+
+    private static List<HashMap> generateRecords(int numberOfRecords) {
+
+        List<HashMap> recordList = new ArrayList<>();
+
+        for (int rows = 0; rows < numberOfRecords; rows++) {
+
+            HashMap<String, String> eventData = new HashMap<>();
+
+            eventData.put("name", "Person" + rows);
```

Comment:
We should include a completely random value (use a UUID) in the event so that we can avoid possible conflicts from tests running in parallel or close to each other. Maybe generate this value in the `@BeforeEach`, make this current method non-static and then use the random "test Id."

Reply:
Again, since each test creates a different log stream, this is not necessary.

Reply:
I see that now. This works.